rich/console.py

```diff
@@ -723,6 +727,7 @@ def __init__(
         self._thread_locals = ConsoleThreadLocals(
             theme_stack=ThemeStack(themes.DEFAULT if theme is None else theme)
         )
+        self._buffer_indices_to_render: Set[int] = {self._thread_locals.buffer_index}
```

Comment:
I think this will break with multiple threads.

You may need to add `_buffer_indices_to_render` to the thread locals.